lib/features/auto-update-subscriptions.js

```diff
@@ -0,0 +1,29 @@
+module.exports = ({rtm, filterFn = () => Promise.resolve(true)}) =>
+  Promise.all([
+    rtm.on('subscription-created', subscription => {
+      if (subscription.allowRead !== true) return;
+
+      filterFn(subscription).then(ok => {
+        if (ok) {
+          rtm.subscribe({channel: subscription.channelId});
+        }
+      });
+    }),
+    rtm.on('subscription-deleted', subscription =>
+      filterFn(subscription).then(ok => {
+        if (ok) {
+          rtm.unsubscribe({channel: subscription.channelId});
+        }
+      })
+    ),
+    rtm.on('subscription-updated', subscription =>
+      filterFn(subscription).then(ok => {
+        if (ok) {
+          const method =
+            subscription.allowRead === true ? 'subscribe' : 'unsubscribe';
+
+          rtm[method]({channel: subscription.channelId});
+        }
+      })
```

Comment:
I think `filterFn` should only be asked for approval of subscribe, not for unsubscribe. Doesn't make sense to reject a unsubscribe because broker will force unsubscribe later anyway.